reVX/least_cost_xmission/cost_creator.py

```diff
@@ -0,0 +1,410 @@
+    def _compute_slope_mult(slope, config=None):
+        """
+        Create slope multiplier raster for a region. Unspecified slopes are
+        left at 1.0
+
+        Parameters
+        ----------
+        slope : numpy.ndarray
+            Slope raster clipped to a region- "Terrain slope in % of grade"
+        config : dict | None
+            Multipliers and slope cut offs for hilly and mountain terrain.
+            Use defaults if None.
+        config['hill_mult'] : float
+            Multiplier for hilly terrain
+        config['mtn_slope'] : float
+            Multiplier for mountainous terrain
+        config['hill_slope'] : float
+            Slope at and above which a cell is classified as hilly.
+        config['mtn_slope'] : float
+            Slope at and above which a cell is classified as mountainous
+
+        Returns
+        -------
+        numpy.ndarray
+            Slope multiplier raster. Minimum value for any cell is 1.
+        """
+        if config is None:
+            config = {}
+        hill_mult = config.get('hill_mult', HILL_MULT)
+        mtn_mult = config.get('mtn_mult', MTN_MULT)
+        hill_slope = config.get('hill_slope', HILL_SLOPE)
+        mtn_slope = config.get('mtn_slope', MTN_SLOPE)
+
+        hilly = (slope >= hill_slope) & (slope < mtn_slope)
+        mountainous = slope >= mtn_slope
+
+        mult_raster = np.ones(slope.shape, dtype=np.float32)
+        mult_raster[hilly] = hill_mult
+        mult_raster[mountainous] = mtn_mult
+
+        return mult_raster
+
+    @staticmethod
+    def _compute_land_use_mult(land_use, multipliers, land_use_classes):
+        """
+        Convert NLCD raster to land use multiplier raster for a single region.
+        Land classes without specified multipliers are left a 1.
+
+        Parameters
+        ----------
+        land_use : numpy.ndarray
+           NLCD land user raster clipped to a region
+        multipliers : dict
+            Multiplier for for land classes, E.g. {'forest': 1.5}
+        land_use_classes : dict
+            NCLD land use codes corresponding to use classes for multipliers
+
+        Returns
+        -------
+        numpy.ndarray
+            Land use multiplier raster. Minimum value for any cell is 1.
+        """
+        mult_raster = np.ones(land_use.shape, dtype=np.float32)
+
+        indices = []  # [(index0, multiplier0, _class_value0), ...]
+        for _class, multiplier in multipliers.items():
+            if _class not in land_use_classes:
+                msg = f'Class {_class} not in land_use_classes: ' +\
+                      f'{land_use_classes}'
+                raise ValueError(msg)
+
+            values = land_use_classes[_class]
+            assert isinstance(values, list)
+
+            for value in values:
+                index = land_use == value
+                indices.append((index, multiplier, value))
+
+        for i in indices:
+            mult_raster[i[0]] = i[1]
+
+        return mult_raster
```

Comment:
Lets add a second class method, could be `run`, `build`, or `create` that would
1) run `build_test_costs`
2) add any extra layers using [`layers_to_h5`](https://github.com/NREL/reVX/blob/adab7d9d2e64ff06cf4ad8438f17952004f34e5d/reVX/utilities/exclusions_converter.py#L508)

Reply:
What do you mean by extra layers?

Reply:
Transmission barrier layer and dist_to_coast layers I think are the only "other" layers we need. i.e none cost layers